src/libmain/progress-bar.cc

```diff
@@ -360,6 +363,28 @@ class ProgressBar : public Logger
     }
 
     std::chrono::milliseconds draw(State & state)
+    {
+        // Call draw() and render if the output has changed.
+
+        // Excessive redrawing is noticable on slow terminals, and it interferes
+        // with text selection in some terminals, including libvte-based terminal
+        // emulators.
+
+        std::optional<std::string> newOutput;
+        auto nextWakeup = draw(state, newOutput);
+        if (newOutput && *newOutput != lastOutput) {
```

Comment:
```suggestion
        if (newOutput != lastOutput) {
```
should work since `std::optional<T>` has overloads for comparing to `T`.

Reply:
The `optional` wasn't really needed, and I've removed it.